`testEvictLeaderSuite.TestEvictLeader` is unstable

## Flaky Test

### Which jobs are failing

```shell
FAIL: scheduler_test.go:248: testEvictLeaderSuite.TestEvictLeader

scheduler_test.go:267:
    testutil.CheckMultiTargetTransferLeader(c, op[0], operator.OpLeader, 1, []uint64{2, 3})
/Users/jmpotato/go/src/github.com/JmPotato/pd/pkg/testutil/operator_check.go:71:
    c.Assert(op.Step(0), DeepEqualsIn, expectedOps)
... obtained operator.TransferLeader = operator.TransferLeader{FromStore:0x1, ToStore:0x3, ToStores:[]uint64{0x3, 0x2}} ("transfer leader from store 1 to store 3")
... expected []interface {} = []interface {}{operator.TransferLeader{FromStore:0x1, ToStore:0x2, ToStores:[]uint64{0x2, 0x3}}, operator.TransferLeader{FromStore:0x1, ToStore:0x3, ToStores:[]uint64{0x2, 0x3}}}

OOPS: 1 passed, 1 FAILED
--- FAIL: Test (0.00s)
FAIL
FAIL	github.com/tikv/pd/server/schedulers	0.435s
FAIL
```

### CI link

https://github.com/tikv/pd/runs/4785883359?check_suite_focus=true

### Reason for failure (if possible)

Related PR: https://github.com/tikv/pd/pull/4401

--- a/server/schedule/operator/builder.go
+++ b/server/schedule/operator/builder.go
@@ -247,6 +247,7 @@ func (b *Builder) SetLeaders(storeIDs []uint64) *Builder {
 	if b.err != nil {
 		return b
 	}
+	sort.Slice(storeIDs, func(i, j int) bool { return storeIDs[i] < storeIDs[j] })
 	for _, storeID := range storeIDs {
 		peer := b.targetPeers[storeID]
 		if peer == nil || core.IsLearner(peer) || b.unhealthyPeers[storeID] != nil {

